jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -148,271 +164,208 @@ def user_api(self):
+            self.spectrum_viewer.set_plot_axes()
             return
-        if not self.flux_unit.choices and self.app.config == 'cubeviz':
-            return
-
-        # various plugins are listening for changes in either flux or sb and
-        # need to be able to filter messages accordingly, so broadcast both when
-        # flux unit is updated. if data was loaded in a flux unit (i.e MJy), it
-        # can be reperesented as a per-pixel surface brightness unit
-        flux_unit = self.flux_unit.selected
-        sb_unit = self._append_angle_correctly(flux_unit, self.angle_unit.selected)
-
-        self.hub.broadcast(GlobalDisplayUnitChanged("flux", flux_unit, sender=self))
-        self.hub.broadcast(GlobalDisplayUnitChanged("sb", sb_unit, sender=self))
-
-        spectral_y = sb_unit if self.spectral_y_type == 'Surface Brightness' else flux_unit
-
-        yunit = _valid_glue_display_unit(spectral_y, self.spectrum_viewer, 'y')
-
-        # update spectrum viewer with new y display unit
-        if self.spectrum_viewer.state.y_display_unit != yunit:
+        try:
             self.spectrum_viewer.state.y_display_unit = yunit
-            self.spectrum_viewer.reset_limits()
-
-            # and broacast that there has been a change in the spectral axis y unit
-            # to either a flux or surface brightness unit, for plugins that specifically
-            # care about this toggle selection
-            self.hub.broadcast(GlobalDisplayUnitChanged("spectral_y", spectral_y, sender=self))
-
-        if not check_if_unit_is_per_solid_angle(self.spectrum_viewer.state.y_display_unit):
-            self.spectral_y_type_selected = 'Flux'
+        except ValueError:
+            # may not be data in the viewer, or unit may be incompatible
+            pass
         else:
-            self.spectral_y_type_selected = 'Surface Brightness'
-
-        # Always send a surface brightness unit to contours
-        if self.spectral_y_type_selected == 'Flux':
-            yunit = self._append_angle_correctly(yunit, self.angle_unit.selected)
-        self._find_and_convert_contour_units(yunit=yunit)
-
-        # for displaying message that PIXAR_SR = 1 if it is not found in the FITS header
-        if (
-            len(self.app.data_collection) > 0
-            and not self.app.data_collection[0].meta.get('PIXAR_SR')
-        ):
-            self.pixar_sr_exists = False
-
-    def _find_and_convert_contour_units(self, msg=None, yunit=None):
-        if not yunit:
-            yunit = self.sb_unit_selected
-
-        if msg is not None:
-            viewers = [self.app.get_viewer(msg.viewer_reference)]
-        else:
-            viewers = self._app._viewer_store.values()
+            self.spectrum_viewer.set_plot_axes()
+            # until we can have upstream automatic limit updating on change
+            # in display units with equivalencies, we'll reset the limits
+            self.spectrum_viewer.reset_limits()
```

Comment:
kept this in because otherwise the limits could wander endlessly.  We'll eventually want to remove this (but it was in main before).